File: java/leetcode_grind/Day718.java
package leetcode_grind;

public class Day718 {
    // https://leetcode.com/problems/string-compression-iii/description/?envType=daily-question&envId=2024-11-04
    static class Solution1 {
        public String compressedString(String word) {
            StringBuilder comp = new StringBuilder("");
            int pos = 0;
            while (pos < word.length()) {
                int consecutiveCount = 0;
                char currentChar = word.charAt(pos);
                while (pos < word.length() && consecutiveCount < 9 && word.charAt(pos) == currentChar) {
                    consecutiveCount++;
                    pos++;
                }
                comp.append(consecutiveCount).append(currentChar);
            }
            return comp.toString();
        }
    }
}

File: src/leetcode_grind/day_718.rs
// https://leetcode.com/problems/string-compression-iii/description/?envType=daily-question&envId=2024-11-04
pub fn compressed_string(word: String) -> String {
    let word = word.chars().collect::<Vec<_>>();
    let mut res = String::new();
    let mut pos = 0;
    while pos < word.len() {
        let mut cnt = 0;
        let c = word[pos];
        while pos < word.len() && cnt < 9 && c == word[pos] {
            cnt += 1;
            pos += 1;
        }
        res.push_str(&format!("{cnt}"));
        res.push(c);
    }
    res
}
